backends/s3/new.go

```diff
@@ -25,6 +26,7 @@ func _new(config *backends.Config) (backends.Client, error) {
 	if !ok {
 		bucket = defaultS3Bucket
 	}
+	bucket = path.TrimModPrefix(bucket)
```

Comment:
Oops, then I misunderstood your explanation in https://github.com/bndw/pick/pull/173#issuecomment-379340110. Does the following patch represent what you want me to change?

```patch
diff --git a/backends/s3/new.go b/backends/s3/new.go
index ba3c582..8306cea 100644
--- a/backends/s3/new.go
+++ b/backends/s3/new.go
@@ -26,12 +26,12 @@ func _new(config *backends.Config) (backends.Client, error) {
 	if !ok {
 		bucket = defaultS3Bucket
 	}
-	bucket = path.TrimModPrefix(bucket)
 
 	key, ok := config.Settings["key"].(string)
 	if !ok {
 		key = defaultS3Key
 	}
+	key = path.TrimModPrefix(key)
 
 	// TODO(bndw): Consider creating the bucket if it does not exist
 
```

I'm a complete S3 beginner, so please forgive my newbness. Should the `key` be prefixed with `safes/` as done for the default safe key?

Reply:
My mistake, I definitely said bucket in that comment.

An S3 bucket can be _loosely_ thought of as a directory that can hold any number of files/folders. I think it's most intuitive for the `--safe` flag to specify _which safe_ to use in the bucket (directory).

I believe the above patch would achieve this.

Reply:
What about the `safes/` prefix? (see above)

Reply:
Not adding the `safes/` prefix makes sense to me, though this may be inconsistent. 

Here's an example of how I configure this today and how I would use the `--safe` flag:

```toml
# ~/.pick/config.toml

[storage]
  type = "s3"

  [storage.settings]
    bucket  = "my-s3-bucket"
    key     = "safes/pick.safe"
    region  = "us-west-2"
    profile = "personal"
```

```sh
# List passwords in my `pick.safe`
pick ls

# List passwords in a safe in a different directory structure
pick --safe work/safes/pick.safe ls
```

Reply:
Updated with the patch posted before. This will however not allow specifying the bucket name, is that something we really want to support?

Reply:
In my usage I wouldn't ever change the bucket, so I think it's fine to only allow that in the config. If someone requests it as a flag we can go from there :)